include/wx/msw/private/tlwgeom.h

```diff
@@ -0,0 +1,126 @@
+///////////////////////////////////////////////////////////////////////////////
+// Name:        wx/msw/private/tlwgeom.h
+// Purpose:     wxMSW-specific wxTLWGeometry class.
+// Author:      Vadim Zeitlin
+// Created:     2018-04-29
+// Copyright:   (c) 2018 Vadim Zeitlin <vadim@wxwidgets.org>
+// Licence:     wxWindows licence
+///////////////////////////////////////////////////////////////////////////////
+
+#ifndef _WX_MSW_PRIVATE_TLWGEOM_H_
+#define _WX_MSW_PRIVATE_TLWGEOM_H_
+
+#include "wx/log.h"
+
+#include "wx/msw/private.h"
+
+// names for MSW-specific options
+#define wxPERSIST_TLW_MAX_X "xmax"
+#define wxPERSIST_TLW_MAX_Y "ymax"
+
+class wxTLWGeometry : public wxTLWGeometryBase
+{
+public:
+    wxTLWGeometry()
+    {
+        wxZeroMemory(m_placement);
+        m_placement.length = sizeof(m_placement);
+    }
+
+    virtual bool Save(const Serializer& ser) const wxOVERRIDE
+    {
+        // For compatibility with the existing saved positions/sizes, use the
+        // same keys as the generic version (which was previously used under
+        // MSW too).
+
+        // Normal position and size.
+        const RECT& rc = m_placement.rcNormalPosition;
+        if ( !ser.SaveField(wxPERSIST_TLW_X, rc.left) ||
+             !ser.SaveField(wxPERSIST_TLW_Y, rc.top) )
+            return false;
+
+        if ( !ser.SaveField(wxPERSIST_TLW_W, rc.right - rc.left) ||
+             !ser.SaveField(wxPERSIST_TLW_H, rc.bottom - rc.top) )
+            return false;
+
+        // Maximized/minimized state.
+        UINT show = m_placement.showCmd;
+        if ( !ser.SaveField(wxPERSIST_TLW_MAXIMIZED, show == SW_SHOWMAXIMIZED) )
+            return false;
+
+        if ( !ser.SaveField(wxPERSIST_TLW_ICONIZED, show == SW_SHOWMINIMIZED) )
+            return false;
+
+        // Maximized window position.
+        const POINT pt = m_placement.ptMaxPosition;
```

Comment:
Shouldn't `pt` be a reference too?

Reply:
It looks like making a copy of a `POINT` is so cheap that it's not worth avoiding it by using a reference...